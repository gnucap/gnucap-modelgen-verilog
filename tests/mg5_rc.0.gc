

attach ../modelgen_0.so

attach ../vams/vsine.so

verilog

`modelgen
module test_rc0(in, out, g);
  inout in, out, g;
  electrical in, out, g;
  resistor #(.r(1)) r(in, out);
  analog begin
    I(g,out) <+ ddt(V(g,out));
  end
endmodule

!make test_rc0.so > /dev/null

attach ./test_rc0.so

verilog

test_rc0 #() dut(in, out, 0);
vsine #(.freq(1), .ampl(1)) v1(in, 0);

print dc v(nodes) iter(0)
dc v -10 10 2

print tran v(nodes) iter(0)
tran .5
status notime
