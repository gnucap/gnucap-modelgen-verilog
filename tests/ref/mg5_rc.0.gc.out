dc v -10 10 2 
   ^ ? what's this?
#           v(in)      v(out)     iter(0)   
 0.         0.         0.         3.        
#Time       v(in)      v(out)     iter(0)   
 0.         0.         0.         2.        
 249.n      1.5645u    390.f      15.       
 786.n      4.9385u    2.136p     3.        
 1.323u     8.3126u    5.694p     3.        
 1.86u      11.687u    11.063p    3.        
 54.742u    343.95u    9.4142n    3.        
 493.68u    0.0031018  765.48n    3.        
 0.0024525  0.015409   18.876u    3.        
 0.008238   0.051737   212.44u    3.        
 0.021179   0.13268    0.0013953  3.        
 0.04398    0.27283    0.0059347  3.        
 0.076553   0.46266    0.017531   3.        
 0.10913    0.63318    0.03453    3.        
 0.15799    0.83747    0.067954   3.        
 0.20684    0.96345    0.10766    3.        
 0.2557     0.99935    0.14933    3.        
 0.30456    0.9418     0.1885     3.        
 0.36971    0.73022    0.22935    3.        
 0.43485    0.39799    0.25047    3.        
 0.5        2.4722u    0.24722    3.        
Gnucap   System status
iterations: op=0, dc=3, tran=71, fourier=0, total=98
transient timesteps: accepted=20, rejected=4, total=24
nodes: user=2, subckt=0, model=1, total=3
dctran density=77.8%, ac density=77.8%
